platforms/jvm/testing-jvm/src/integTest/groovy/org/gradle/testing/NonClassBasedTestingIntegrationTest.groovy

```diff
@@ -0,0 +1,498 @@
+                        testDefinitionDirs.from(project.layout.projectDirectory.file("$badPath"))
+                    }
+                }
+            }
+        """
+
+        // Write test defs to default dir (which is still scanned), not badPath, needed to avoid "no sources" skip
+        writeTestDefinitions()
+
+        when:
+        fails("test")
+
+        then:
+        failureCauseContains("Test definitions directory does not exist: " + testDirectory.file(badPath).absolutePath)
+    }
+
+    def "non-directory test definitions directory fails"() {
+        def badPath = "src/test/i-dont-exist.txt"
+        file(badPath).createFile()
+
+        given:
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+
+            ${mavenCentralRepository()}
+
+            testing.suites.test {
+                ${setupSuiteWithEngineFixture()}
+
+                targets.all {
+                    testTask.configure {
+                        scanForTestDefinitions = true
+                        testDefinitionDirs.setFrom(project.layout.projectDirectory.file("$badPath"))
+                    }
+                }
+            }
+        """
+
+        when:
+        fails("test")
+
+        then:
+        failureCauseContains("Test definitions directory is not a directory: " + testDirectory.file(badPath).absolutePath)
+    }
+
+    @Requires(UnitTestPreconditions.NotWindows)
+    def "non-readable test definitions directory fails"() {
+        def badPath = "src/test/i-cant-be-read"
+        // create the directory and make it non-readable at the filesystem level
+        def dir = file(badPath).createDir()
+
+        // try to remove read permission for everyone (owner=false, ownerOnly=false)
+        // fall back to single-arg version if the two-arg form isn't supported/has no effect
+        if (!dir.setReadable(false, false)) {
+            dir.setReadable(false)
+        }
+
+        given:
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+
+            ${mavenCentralRepository()}
+
+            testing.suites.test {
+                ${setupSuiteWithEngineFixture()}
+
+                targets.all {
+                    testTask.configure {
+                        scanForTestDefinitions = true
+                        testDefinitionDirs.setFrom(project.layout.projectDirectory.file("$badPath"))
+                    }
+                }
+            }
+        """
+
+        when:
+        fails("test")
+
+        then:
+        failureCauseContains("Cannot access input property 'candidateDefinitionDirs' of task ':test'. Accessing unreadable inputs or outputs is not supported.")
+        failureCauseContains("java.nio.file.AccessDeniedException: ${dir.absolutePath}")
+
+        cleanup:
+        // restore read permission for cleanup
+        dir.setReadable(true, false)
+    }
+
+    def "non-readable files in test definitions directory fails"() {
```

Comment:
Same thing here as above, this is testing the execution engine and not something specific to Test